crates/red_knot_python_semantic/src/types.rs

```diff
@@ -3281,6 +3489,18 @@ impl<'db> Type<'db> {
             Type::Callable(_) => KnownClass::Type.to_instance(db),
             Type::ModuleLiteral(_) => KnownClass::ModuleType.to_class_literal(db),
             Type::Tuple(_) => KnownClass::Tuple.to_class_literal(db),
+
+            Type::TypeVar(typevar) => match typevar.bound_or_constraints(db) {
+                None => KnownClass::Object.to_class_literal(db),
+                Some(TypeVarBoundOrConstraints::UpperBound(bound)) => bound.to_meta_type(db),
+                Some(TypeVarBoundOrConstraints::Constraints(constraints)) => {
+                    // TODO: This should create ClassLiterals, not SubclassOfTypes, for each
+                    // element, since constraints must be specialized to those specific types, not
+                    // to subclasses of those types.
+                    constraints.map(db, |constraint| constraint.to_meta_type(db))
```

Comment:
The TODO comment I do feel we might need here is that I don't think having this return a union type is really correct. I think it should rather be a synthetic typevar constrained to the meta-types of each constraint.

I'm not sure how much this matters in practice, so totally fine with it being a TODO.

Reply:
After further discussion this morning, I think this is one of those cases where returning the union is not wrong, just imprecise, and probably fine.